src/librustc_mir/interpret/visitor.rs

```diff
@@ -130,6 +130,9 @@ macro_rules! make_value_visitor {
         pub trait $visitor_trait_name<'a, 'mir, 'tcx: 'mir+'a, M: Machine<'a, 'mir, 'tcx>>: Sized {
             type V: Value<'a, 'mir, 'tcx, M>;
 
+            /// Controls whether aggregate fields will be sorted by offset.
+            const WANT_FIELDS_SORTED: bool;
```

Comment:
what's this about?

Reply:
miri needs to visit the fields of a struct sorted by their offset. That is the best way I found so far to implement something that visit *everything* (including padding) that's not inside an `UnsafeCell`: You visit all the `UnsafeCell`, and since you get them sorted, you know that everything between the last `UnsafeCell` and this one is frozen.

I just realized, however, that I could also first collect all the `UnsafeCell` and then sort after the fact. So if you prefer I can remove this from the visitor again and implement this entirely in miri.

Reply:
No it's fine (if changed to a method and with the above as docs on the method)